### ISSUE TITLE
Support Mutations in storage Memory.

### ISSUE THREAD

**Issue by qoega** [ClickHouse/ClickHouse#9117]
**Use case**
```(sql)
ALTER TABLE test
    UPDATE test = '1' WHERE 1


Received exception from server (version 20.2.1):
Code: 48. DB::Exception: Received from localhost:9000. DB::Exception: Mutations are not supported by storage Memory.
```

**Comment by alexey-milovidov** [comment]
The use case is unclear.
But it may be a nice and easy task to do.

**Comment by icoleslaw** [comment]
Just ran into this while testing out some queries on a temporary table.

**Comment by jakesyl** [comment]
Use case is using `memory` ENGINE in the way described in #11459 (constant recomputes on insert)

**Comment by flipper86** [comment]
I tried to achieve the same, couldn't change the system.zookeeper table. I'm currently stuck because I'm trying to get create a replicated table, but I get an eerror saying the zookeeper path already exists, yet I cn't do any other operation because there is no table (no data nor metadata in clickhouse). So the zookeeper path is present but the table itself is not. Any suggestion on how to tackle this?

**Comment by tavplubix** [comment]
> I tried to achieve the same, couldn't change the system.zookeeper table. I'm currently stuck because I'm trying to get create a replicated table, but I get an error saying the zookeeper path already exists, yet I can't do any other operation because there is no table (no data nor metadata in clickhouse). So the zookeeper path is present but the table itself is not, and I need to create this table with THAT zookeeper path. Any suggestion on how to tackle this? Thanks

There is no way to change `system.zookeeper`, it's not even `Memory` table, so mutations support in `Memory` engine will not help.

Try `SYSTEM DROP REPLICA 'some_name' FROM ZKPATH '/path/to/table/in/zk';` to remove that path or remove it manually with some zookeeper client.

**Comment by flipper86** [comment]
> > I tried to achieve the same, couldn't change the system.zookeeper table. I'm currently stuck because I'm trying to get create a replicated table, but I get an error saying the zookeeper path already exists, yet I can't do any other operation because there is no table (no data nor metadata in clickhouse). So the zookeeper path is present but the table itself is not, and I need to create this table with THAT zookeeper path. Any suggestion on how to tackle this? Thanks
> 
> There is no way to change `system.zookeeper`, it's not even `Memory` table, so mutations support in `Memory` engine will not help.
> 
> Try `SYSTEM DROP REPLICA 'some_name' FROM ZKPATH '/path/to/table/in/zk';` to remove that path or remove it manually with some zookeeper client.

Thanks for the suggestions. I first tried to run the system drop command, but I got error saying:

`Code: 305. DB::Exception: Received from localhost:9000. DB::Exception: Can't remove replica: replica1, because it's active.`

so I deleted the path manually with the zookeeper client zkCli.sh, and everything worked again. 
Thanks!

**Comment by ucasFL** [comment]
@abyss7 Hi, I have implemented this feature in [#15127](https://github.com/ClickHouse/ClickHouse/pull/15127), maybe you can help review it.
